spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -256,12 +262,62 @@ public static void applyQueryOptions(StructuredQuery.Builder builder, DatastoreQ
 						: StructuredQuery.OrderBy.Direction.ASCENDING);
 	}
 
-	private Entity convertToEntityForSave(Object entity) {
-		Builder builder = Entity.newBuilder(getKey(entity, true));
+	private Entity convertToEntityForSave(Object entity, Key... ancestors) {
+		if (ancestors != null) {
+			for (Key ancestor : ancestors) {
+				validateKey(entity, keyToPathElement(ancestor));
+			}
+		}
+		Builder builder = Entity.newBuilder(getKey(entity, true, ancestors));
 		this.datastoreEntityConverter.write(entity, builder);
+		saveDescendents(entity, getKey(entity, true, ancestors));
 		return builder.build();
 	}
 
+	private void saveDescendents(Object entity, Key key) {
+		DatastorePersistentEntity datastorePersistentEntity = this.datastoreMappingContext
+				.getPersistentEntity(entity.getClass());
+		datastorePersistentEntity.doWithDescendantProperties(
+				(PersistentProperty persistentProperty) -> {
+					//Convert and write descendants, applying ancestor from parent entry
+					PersistentPropertyAccessor accessor = datastorePersistentEntity.getPropertyAccessor(entity);
+					Object val = accessor.getProperty(persistentProperty);
+
+					//Check if property is a non-null array
+					if (val != null && val.getClass().isArray() && val.getClass() != byte[].class) {
+						//if a propperty is an array, convert it to list
+						val = CollectionUtils.arrayToList(val);
+					}
+					if (val instanceof Iterable) {
```

Comment:
Is it already verified that it's collection-like? If not, we should also have an `else` throwing an exception.